### PR TITLE
WeldContext Bindings for weld-java

### PR DESCRIPTION
**Pull request by sppalkia**
Requires https://github.com/weld-project/weld/pull/381 to be merged first.

### CODE REVIEW

File: src/main/scala/weld/WeldContext.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+object WeldContext {
+  /**
+   * Initialize a [[WeldContext]] using the default configuration.
+   */
+  def init(): WeldContext = {
+    val conf = new WeldConf
+    try init(conf) finally {
+      conf.close()
```
Comment (abehm) [inline src/main/scala/weld/WeldContext.scala:29]: Why close the conf?

**Review by abehm — COMMENTED** [review]
lgtm

File: src/main/scala/weld/WeldContext.scala (hunk 1 above)
Comment (sppalkia) [reply src/main/scala/weld/WeldContext.scala:29]: This method is just initializing a default context (similar to what's happening in WeldModule.scala) -- the `init` below supports passing in a `conf`.

Comment (abehm) [reply src/main/scala/weld/WeldContext.scala:29]: Got it I misunderstood. I thought the conf was held onto within the context, but we just parse out some info from the conf and then it's safe to throw away.

**Comment by abehm** [comment]
lgtm

**Pull request merged**